### PR TITLE
Qt unmark

### PR DESCRIPTION
**Pull request by pirogronian**
I'm unhappy about buttons layout, but can't find more sensible...

### CODE REVIEW

File: src/celestia/qt/qtdeepskybrowser.cpp [hunk 1]
```diff
@@ -629,6 +634,24 @@ void DeepSkyBrowser::slotMarkSelected()
...
+    for (int row = 0; row < nRows; row++)
+    {
+        if (sm->isRowSelected(row, QModelIndex()))
+        {
+            DeepSkyObject* dso = dsoModel->itemAtRow((unsigned int) row);
+            if (dso != nullptr)
+                universe->unmarkObject(Selection(dso), 1);
+        } // isRowSelected
+    } // for
+}
```
Comment (375gnu) [inline src/celestia/qt/qtdeepskybrowser.cpp]: The same as for CelestialBrowser::slotUnmarkSelected()

File: src/celestia/qt/qtsolarsystembrowser.cpp [hunk 2]
```diff
@@ -870,6 +875,20 @@ void SolarSystemBrowser::slotMarkSelected()
     }
 }
 
+void SolarSystemBrowser::slotUnmarkSelected()
+{
+    QItemSelectionModel* sm = treeView->selectionModel();
+    QModelIndexList rows = sm->selectedRows();
+    Universe* universe = appCore->getSimulation()->getUniverse();
+    string label;
```
Comment (375gnu) [inline src/celestia/qt/qtsolarsystembrowser.cpp]: unused variable

File: src/celestia/qt/qtsolarsystembrowser.cpp [hunk 3]
```diff
@@ -870,6 +875,20 @@ void SolarSystemBrowser::slotMarkSelected()
     }
 }
 
+void SolarSystemBrowser::slotUnmarkSelected()
+{
+    QItemSelectionModel* sm = treeView->selectionModel();
+    QModelIndexList rows = sm->selectedRows();
```
Comment (375gnu) [inline src/celestia/qt/qtsolarsystembrowser.cpp]: QModelIndexList rows = treeView->selectionModel()->selectedRows();

File: src/celestia/qt/qtcelestialbrowser.cpp [hunk 4]
```diff
@@ -721,6 +726,24 @@ void CelestialBrowser::slotMarkSelected()
...
+    for (int row = 0; row < nRows; row++)
+    {
+        if (sm->isRowSelected(row, QModelIndex()))
+        {
+            Selection sel = starModel->itemAtRow((unsigned int) row);
+            if (sel.star() != nullptr)
+                universe->unmarkObject(sel, 1);
+        } // isRowSelected
+    } // for
+}
```
[the remark this answers is not in the document]
Comment (pirogronian) [reply src/celestia/qt/qtcelestialbrowser.cpp]: In fact I just adopted `slotSelected()`. Furthermore, your proposition took more effort I predicted, but finally works.

File: src/celestia/qt/qtselectionpopup.cpp [hunk 5]
```diff
@@ -173,6 +173,13 @@ SelectionPopup::SelectionPopup(const Selection& sel,
     QMenu* markMenu = createMarkMenu();
     addMenu(markMenu);
 
+    if (appCore->getSimulation()->getUniverse()->isMarked(selection, 1))
+    {
+        QAction *action = new QAction(_("Unmark"), this);
```
[the remark this answers is not in the document]
Comment (pirogronian) [reply src/celestia/qt/qtselectionpopup.cpp]: I forgot to commit this file... [facepalm]

**Pull request merged**